lib/std/ascii.zig

```diff
@@ -267,18 +267,24 @@ test "ASCII character classes" {
 
 /// Writes a lower case copy of `ascii_string` to `output`.
 /// Asserts `output.len >= ascii_string.len`.
+/// `output` and `ascii_string` may be the same.
 pub fn lowerString(output: []u8, ascii_string: []const u8) []u8 {
-    std.debug.assert(output.len >= ascii_string.len);
-    for (ascii_string, 0..) |c, i| {
-        output[i] = toLower(c);
+    const real_output = output[0..ascii_string.len];
```

Comment:
you removed the assert, but didn't mention why. You also didn't remove the comment saying it will assert. The latter across a few of the places you deleted the assert.

Reply:
Ahh, I think I misunderstood exactly what having that line would do, and misattributed the segv I was getting, which only happens when the assert is present.